src/app/dashboard/hex-map/hex-map.component.ts

```diff
@@ -130,6 +170,30 @@ export class HexMapComponent implements OnInit, OnDestroy {
                     }
                 }
             );
+
+            // count the number ot hex-cells
+            const num_of_cells = result.length - 1 // first row is the CSV header
+            // Only auto-switch if user hasn't manually overridden
+            if (this.currentResolution === 3 &&
+                num_of_cells * (7 * 7 * 7) < this.MAX_HEX_CELL &&
+                !this.manualResolutionOverride) {
+
+                this.canToggleResolution = true;
+                this.currentResolution = 6;
+                const reqParams = {
+                    hashtag: this.relevantState().hashtag,
+                    start: this.relevantState().start,
+                    end: this.relevantState().end,
+                    countries: this.relevantState().countries,
+                    topic: this.relevantState().active_topic,
+                    resolution: 6
+                }
+                this.updateLayer(reqParams)
```

Comment:
Id suggest storing the r3 result, not overriding it immediately with r6 - this would save us from requesting from the API again and again on toggle use. Probably just a minor performance impact tho, so if its awkward to code it like that its fine as is as well :)